src/Symfony/Component/Messenger/Bridge/Amqp/Tests/Transport/AmqpSenderTest.php

```diff
@@ -118,4 +120,36 @@ public function testItThrowsATransportExceptionIfItCannotSendTheMessage()
         $sender = new AmqpSender($connection, $serializer);
         $sender->send($envelope);
     }
+
+    public function testSendWithDelay()
+    {
+        $envelope = (new Envelope(new DummyMessage('Oy')))->with(new DelayStamp(1000));
+        $encoded = ['body' => '...', 'headers' => ['type' => DummyMessage::class]];
+
+        $serializer = $this->createMock(SerializerInterface::class);
+        $serializer->method('encode')->with($envelope)->willReturnOnConsecutiveCalls($encoded);
+
+        $connection = $this->createMock(Connection::class);
+        $connection->expects($this->once())->method('publish')->with($encoded['body'], $encoded['headers'], 1000);
+
+        $sender = new AmqpSender($connection, $serializer);
+        $sender->send($envelope);
+    }
```

Comment:
Not really a part of this feature, but there was no test for DelayStamp in AMQP transport.